### PR TITLE
Mobile redirection doc - WIP

### LINKED ISSUE #1367: Mobile Redirection: New Doc Proposal
## Title
Mobile Redirection

## Outline
1. Responsive
1. CDN Redirection

## Path
(e.g. `source/docs/articles/sites/code` or `source/docs/articles/wordpress`)

### PR DESCRIPTION
**Pull request by nataliejeremy**
Closes #1367 

## Remaining Work
- [x] Work with @newtoid to test and complete the doc
- [x] Copy edit for style and grammar

### CODE REVIEW

**Comment by nataliejeremy** [comment]
@newtoid @rachelwhitton please review, then merge.

**Comment by nataliejeremy** [comment]
@rachelwhitton pls review and merge--thx!

**Comment by rachelwhitton** [comment]
Suggest the following TODO:
- [ ] Include responsive javascript alternatives 
- [ ] Test cookie with Varnish `__cf_mob_redir = 0; domain=.example.com` (for disabling mobile redirect when users want to use the desktop site)
- [ ] Update existing docs. [Device Detection](https://pantheon.io/docs/articles/sites/varnish/caching-advancedtopics/#device-detection) states:
 
 > We do not recommend building separate mobile sites or using cookies that are passed to the backend for mobile theme detection and configuration. This will cause issues scaling requests within your site in case of any load or traffic spikes, as it requires at least the initial hit to make it to the backend before anonymous traffic can be cached by Varnish. If you receive more uncached visitors than your Nginx and PHP processes, it can result in timeouts and server errors.

File: source/docs/articles/sites/domains/mobile-redirection.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+category:
+  - developing
+keywords: dns, cloudflare, mobile, redirect-incoming-requests, mobile tools
+---
+This article walks you though setting up a mobile redirect service through CloudFlare to automatically redirect visitors using mobile device visitors to a mobile optimized home page.
+
+## Before You Begin
+
+You'll need to have:  
+- A [CloudFlare account](https://www.cloudflare.com/a/sign-up)
```
Comment (rachelwhitton) [inline source/docs/articles/sites/domains/mobile-redirection.md]: Suggest clarifying assumed config like a domain added to the live environment with DNS already configured using CloudFlare as a provider. Also, an already built and optimized mobile home page.

File: source/docs/articles/sites/domains/mobile-redirection.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- A [CloudFlare account](https://www.cloudflare.com/a/sign-up)  
+- DNS information for the domain you want to redirect
+
+<div class="alert alert-info" role="alert">
+<h4>Note</h4>
+This will only work for your Live domain. If you want to test it for the Dev or Test environments, you will have to point the Live domain name to Dev or Test on Pantheon.</div>
+
+## Add Your Mobile Domain To Your Pantheon Site
+1. From the Site Dashboard, go to your Live environment and select **Domains/HTTPS**.
+2. Add the domain or subdomain you want to use to serve the mobile version of your site (e.g. m.example.com).
```
Comment (rachelwhitton) [inline source/docs/articles/sites/domains/mobile-redirection.md]: Suggest clarifying that the root and www domain should already exist - the mobile subdomain is in addition to already live domains.

File: source/docs/articles/sites/domains/mobile-redirection.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+## Add Your Mobile Domain To Your Pantheon Site
+1. From the Site Dashboard, go to your Live environment and select **Domains/HTTPS**.
+2. Add the domain or subdomain you want to use to serve the mobile version of your site (e.g. m.example.com).  
+You will see a new recommended DNS record.
+
+## Enable Mobile Redirection on CloudFlare
+1. Log in to your CloudFlare account.
+2. Go to the domain you want to set up a mobile redirect for.
+3. Click the **DNS** icon at the top of the page.
+4. Add the newly created mobile domain or subdomain and recommended DNS record.
```
Comment (rachelwhitton) [inline source/docs/articles/sites/domains/mobile-redirection.md]: Suggest removing "domain" and aligning doc contents on the suggestion to use a subdomain (e.g. `m.example.com`) for mobile redirection.

File: source/docs/articles/sites/domains/mobile-redirection.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+5. Click the **Speed** icon at top of the page.
+6. Scroll down to the **Mobile Redirect** section.
+7. Set up the mobile domain as needed.
```
Comment (rachelwhitton) [inline source/docs/articles/sites/domains/mobile-redirection.md]: Suggest covering how to set up the mobile domain. There are 2 configuration options within CF's Mobile Redirect section. One for selecting the mobile site's dns entry. The other selection either drops the path for the URL request or it keeps the path - which do we recommend and why?

**Pull request merged**